### DISCUSSION TITLE
Current state vs Target state

### DISCUSSION THREAD

**Discussion by danilkorotkov** [HomeSpan/HomeSpan#57] [Q&A]
Send notification, no dashboard changes 
`gate->CurrentDoorState-> setVal( !gate->CurrentDoorState->getVal() );`

then
`gate->TargetDoorState-> setVal( gate->CurrentDoorState->getVal() );`
makes dashboard changes

but if Target-> setVal called before Current-> setVal, no notifications but the icon is changed

what is the right way to use them?

**Comment by HomeSpan** [comment]
Sorry, can you be more specific by "no notifications".  If you set the Log Messages to Level 2, does the Serial Monitor show that HomeSpan is sending Event Messages?  Also, note that CurrentDoorState is not a boolean.  Its values range from 0-4 (per the HAP documentation).

**Reply by danilkorotkov** [reply]
i mean push notification in home.app

> Also, note that CurrentDoorState

just for test

**Comment by HomeSpan** [comment]
I'm not sure how push notification logic works in the home app itself - it sits outside of the HAP/HomeKit protocol and thus Apple does not provide anything about this in their HAP documentation.

The solution is to test out different combinations and see how the Home App responds, which is exactly what you did.  Based on what you observed, my guess is that the Home App might be looking for some of the intermediate CurrentDoorStates, such as "opening" and "closing" as opposed go just "open" and "close".  These are explained in the HAP document.
